rules/CodeQuality/Rector/Class_/DynamicDocBlockPropertyToNativePropertyRector.php

```diff
@@ -0,0 +1,225 @@
+        $classPhpDocInfo = $this->phpDocInfoFactory->createFromNode($node);
+        if ($classPhpDocInfo instanceof PhpDocInfo) {
+            $propertyPhpDocTagNodes = $classPhpDocInfo->getTagsByName('property');
+
+            $newProperties = $this->createNewPropertyFromPropertyTagValueNodes($propertyPhpDocTagNodes, $node);
+
+            // remove property tags
+            foreach ($propertyPhpDocTagNodes as $propertyPhpDocTagNode) {
+                // remove from docblock
+                $this->phpDocTagRemover->removeTagValueFromNode($classPhpDocInfo, $propertyPhpDocTagNode);
+            }
+
+            // merge new properties to start of the file
+            $node->stmts = array_merge($newProperties, $node->stmts);
+
+            // update doc info
+            $this->docBlockUpdater->updateRefactoredNodeWithPhpDocInfo($node);
+        }
+
+        return $node;
+    }
+
+    public function provideMinPhpVersion(): int
+    {
+        return PhpVersionFeature::DEPRECATE_DYNAMIC_PROPERTIES;
+    }
+
+    /**
+     * @param PhpDocTagNode[] $propertyPhpDocTagNodes
+     * @return Property[]
+     */
+    private function createNewPropertyFromPropertyTagValueNodes(array $propertyPhpDocTagNodes, Class_ $class): array
+    {
+        $newProperties = [];
+
+        foreach ($propertyPhpDocTagNodes as $propertyPhpDocTagNode) {
+            // add explicit native property
+            $propertyTagValueNode = $propertyPhpDocTagNode->value;
+            if (! $propertyTagValueNode instanceof PropertyTagValueNode) {
+                continue;
+            }
+
+            $propertyName = ltrim($propertyTagValueNode->propertyName, '$');
+
+            if ($this->isPromotedProperty($class, $propertyName)) {
+                continue;
+            }
+
+            // is property already defined?
+            if ($class->getProperty($propertyName)) {
+                // improve exising one type if needed
+
+                $existingProperty = $class->getProperty($propertyName);
+                if ($existingProperty->type !== null) {
+                    continue;
+                }
+
+                $defaultValue = $existingProperty->props[0]->default;
+                $isNullable = $defaultValue instanceof Expr && $this->valueResolver->isNull($defaultValue);
+
+                $existingProperty->type = $this->typedPropertyFactory->createPropertyTypeNode(
+                    $propertyTagValueNode,
+                    $class,
+                    $isNullable,
+                );
+
+                continue;
+            }
+
+            $newProperties[] = $this->typedPropertyFactory->createFromPropertyTagValueNode(
+                $propertyTagValueNode,
+                $class,
+                $propertyName
+            );
+        }
+
+        return $newProperties;
+    }
+
+    private function shouldSkipClass(Class_ $class): bool
+    {
+        // skip magic
+        $getClassMethod = $class->getMethod('__get');
```

Comment:
This check should from ClassReflection since __get can be in parent class

Reply:
In such case, the attribute would be defined in parent class only.

> Child classes automatically inherit the #[AllowDynamicProperties] attribute from parent class.

Let's keep it simple for now